sonic-stormond/scripts/stormond

```diff
@@ -0,0 +1,379 @@
+                dynamic_kvp_dict["reserved_blocks"] = self.storage.devices[storage_device].get_reserved_blocks()
+
+                #import pdb; pdb.set_trace()
+                dynamic_kvp_dict["total_fsio_reads"], dynamic_kvp_dict["total_fsio_writes"] = self._reconcile_fsio_rw_values(dynamic_kvp_dict, storage_device)
+
+                self.log_info("Storage Device: {}, Firmware: {}, health: {}%, Temp: {}C, FS IO Reads: {}, FS IO Writes: {}".format(\
+                storage_device, dynamic_kvp_dict["firmware"], dynamic_kvp_dict["health"], dynamic_kvp_dict["temperature"], dynamic_kvp_dict["total_fsio_reads"],dynamic_kvp_dict["total_fsio_writes"]))
+                self.log_info("Latest FSIO Reads: {}, Latest FSIO Writes: {}".format(dynamic_kvp_dict["latest_fsio_reads"], dynamic_kvp_dict["latest_fsio_writes"]))
+                self.log_info("Disk IO Reads: {}, Disk IO Writes: {}, Reserved Blocks: {}".format(dynamic_kvp_dict["disk_io_reads"], dynamic_kvp_dict["disk_io_writes"], \
+                dynamic_kvp_dict["reserved_blocks"]))
+
+                # update Storage Device Status to DB
+                self.update_storage_info_status_db(storage_device, self.dynamic_fields, dynamic_kvp_dict)
+
+            except Exception as ex:
+                self.log_info("get_dynamic_fields() failed with: {}".format(str(ex)))
+
+    
+   # Override signal handler from DaemonBase
+    def signal_handler(self, sig, frame):
+        FATAL_SIGNALS = [signal.SIGINT, signal.SIGTERM]
+        NONFATAL_SIGNALS = [signal.SIGHUP]
+
+        global exit_code
+
+        if sig in FATAL_SIGNALS:
+            self.log_info("Caught signal '{}'".format(SIGNALS_TO_NAMES_DICT[sig]))
+            
+            self.log_info("Syncing latest procfs reads and writes to disk")
+            self._sync_fsio_rw_json()
+
+            self.log_info(" Exiting.")
+
+            # Make sure we exit with a non-zero code so that supervisor will try to restart us
+            exit_code = 128 + sig  
+            self.stop_event.set()
+        elif sig in NONFATAL_SIGNALS:
+            self.log_info("Caught signal '{}' - ignoring...".format(SIGNALS_TO_NAMES_DICT[sig]))
+        else:
+            self.log_info("Caught unhandled signal '{}' - ignoring...".format(SIGNALS_TO_NAMES_DICT[sig]))
+
+    # Main daemon logic
+    def run(self):
+
+        if self.stop_event.wait(self.timeout):
+            # We received a fatal signal
+            return False
+
+        # Check if time elapsed since init is > fsstats_sync_interval OR
+        # If sync interval has elapsed or if difference in elapsed_time and sync interval is less than polling interval
+
+        # If so, sync the appropriate fields to FSIO JSON file
+        
+        elapsed_time = time.time() - self.fsio_sync_time
+        if (elapsed_time > self.fsstats_sync_interval) or ((self.fsstats_sync_interval - elapsed_time) < self.timeout):
```

Comment:
If i understand, the need for second condition is to pre sync if the sync time falls within next timeout. Why is this required? I believe there should not be any problem if it is delayed until the timeout.

Reply:
There is no problem per se. I felt that it would make intuitive sense that if time left before sync needs to happen is < time remaining for the next polling, we can sync now and push the next sync time back.

Reply:
If there is no other discussion reg. this, can I resolve this discussion?

Reply:
Resolving.